src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -154,35 +155,54 @@ function useChatUtilsHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatUtilsHook {
   const helperSaveChannelDetail = async (channel: ChannelList, response: any) => {
     if (!localDb) return;
 
-    queue.addPriorityJob({
-      operationLabel: DatabaseOperationLabel.FetchChannelDetail_SaveAllChat,
-      id: channel?.id,
-      priority: QueueJobPriority.MEDIUM,
-      task: async () => {
-        const saveChatPromises = response?.messages?.map((message) => {
-          return new Promise((resolve) => {
-            // Skip if message_type is notification-deleted
-            if (message?.message_type === 'notification-deleted') return resolve(true);
-
-            const chatMessage = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
-            chatMessage.save(localDb);
-            resolve(true);
-          });
+    if (Platform.OS === 'android') {
+      const saveChatPromises = response?.messages?.map((message) => {
+        return new Promise((resolve) => {
+          // Skip if message_type is notification-deleted
+          if (message?.message_type === 'notification-deleted') return resolve(true);
+
+          const chatMessage = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+          chatMessage.save(localDb);
+          resolve(true);
         });
+      });
 
-        await Promise.all(saveChatPromises);
-      }
-    });
+      await Promise.all(saveChatPromises);
+    } else {
+      queue.addPriorityJob({
+        operationLabel: DatabaseOperationLabel.FetchChannelDetail_SaveAllChat,
+        id: channel?.id,
+        priority: QueueJobPriority.MEDIUM,
+        task: async () => {
+          const saveChatPromises = response?.messages?.map((message) => {
+            return new Promise((resolve) => {
+              // Skip if message_type is notification-deleted
+              if (message?.message_type === 'notification-deleted') return resolve(true);
+
+              const chatMessage = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+              chatMessage.save(localDb);
+              resolve(true);
+            });
+          });
 
-    queue.addPriorityJob({
-      operationLabel: DatabaseOperationLabel.FetchChannelDetail_RefreshChannelList,
-      id: channel?.id,
-      priority: QueueJobPriority.LOW,
-      task: async () => {
-        refresh('channelList');
-      },
-      forceAddToQueue: true
-    });
+          await Promise.all(saveChatPromises);
+        }
+      });
+    }
+
+    if (Platform.OS === 'android') {
+      refresh('channelList');
+    } else {
+      queue.addPriorityJob({
+        operationLabel: DatabaseOperationLabel.FetchChannelDetail_RefreshChannelList,
+        id: channel?.id,
+        priority: QueueJobPriority.LOW,
+        task: async () => {
+          refresh('channelList');
+        },
+        forceAddToQueue: true
+      });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes here introduce platform-specific conditional checks for saving chat messages and refreshing the channel list. For Android, tasks are processed directly, while on other platforms, tasks are added to a priority job queue. This seems like an optimization for Android devices, but it's important to ensure that this doesn't introduce any inconsistencies between platforms. 

One potential issue I see here is error handling. If there's an error during the execution of `chatMessage.save(localDb);` or `refresh('channelList');`, it won't be caught and handled properly. It would be better to wrap these operations in try-catch blocks to handle any potential errors.

```diff
    if (Platform.OS === 'android') {
      const saveChatPromises = response?.messages?.map((message) => {
        return new Promise((resolve, reject) => {
          // Skip if message_type is notification-deleted
          if (message?.message_type === 'notification-deleted') return resolve(true);

          const chatMessage = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
          try {
            chatMessage.save(localDb);
            resolve(true);
          } catch (error) {
            reject(error);
          }
         });
      });
 
      await Promise.all(saveChatPromises);
    } else {
      // ... existing code ...
    }

    if (Platform.OS === 'android') {
      try {
        refresh('channelList');
      } catch (error) {
        console.error('Error refreshing channel list:', error);
      }
    } else {
      // ... existing code ...
    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->